auth/token-managers/iam-token-manager.ts

```diff
@@ -143,6 +161,10 @@ export class IamTokenManager extends JwtTokenManager {
       }
     };
 
+    if (this.scope && this.scope !== '') {
```

Comment:
The second check is redundant - if it is an empty string `this.scope` will return `false`
```suggestion
    if (this.scope) {
```